packages/e2e-tests/src/specs/editor/taxonomy.js

```diff
@@ -0,0 +1,102 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import {
+  createNewStory,
+  publishStory,
+  withUser,
+} from '@web-stories-wp/e2e-test-utils';
+
+describe('Taxonomy', () => {
+  const goToAndExpandTaxonomyPanel = async () => {
+    await expect(page).toClick('li[role="tab"]', { text: 'Document' });
+    await expect(page).toMatch('Categories and Tags');
+
+    const taxonomyPanel = await page.$(
+      'button[aria-label="Categories and Tags"]'
+    );
+
+    const isCollapsed = await page.evaluate(
+      (button) => button.getAttribute('aria-expanded') == 'false',
+      taxonomyPanel
+    );
+    if (isCollapsed) {
+      await taxonomyPanel.click();
+    }
+  };
+
+  const addChildCategory = async ({ parent, child }) => {
+    await expect(page).toClick(
+      'button[data-testid="expand_add_new_hierarchical_term"]'
+    );
+    await page.type('input[name="New Category Name"]', child);
+    await expect(page).toClick('button[aria-label="Parent Category"]');
+    await expect(page).toClick('li[role="option"]', { text: parent });
+    await expect(page).toClick(
+      'button[data-testid="submit_add_new_hierarchical_term"]'
+    );
+  };
+
+  // TODO  https://github.com/google/web-stories-wp/issues/9149
+  // eslint-disable-next-line jest/no-disabled-tests
+  it.skip('should be able to add new categories', async () => {
+    await createNewStory();
+    await goToAndExpandTaxonomyPanel();
+
+    await expect(page).toMatch('Add New Category');
+    await expect(page).toClick(
+      'button[data-testid="expand_add_new_hierarchical_term"]'
+    );
+    await page.waitForSelector('input[name="New Category Name"]');
+    // add a parent category
+    await page.type('input[name="New Category Name"]', 'music genres');
+    await expect(page).toClick(
+      'button[data-testid="submit_add_new_hierarchical_term"]'
+    );
+
+    // add some child categories
+    await addChildCategory({ parent: 'music genres', child: 'rock' });
+    await addChildCategory({ parent: 'music genres', child: 'jazz' });
+    await addChildCategory({ parent: 'music genres', child: 'industrial' });
+    await addChildCategory({ parent: 'music genres', child: 'electro-pop' });
+    await addChildCategory({ parent: 'music genres', child: 'funk' });
+    // Save terms so that they are available to contributor
+    await publishStory();
+  });
+  it.todo('should be able to add new tags and existing tags');
+
+  describe('Contributor User', () => {
+    withUser('contributor', 'password');
+    // right now enabling the taxonomy experiment is needed to make this test at all worthwhile
+    // TODO https://github.com/google/web-stories-wp/issues/8833
+    // eslint-disable-next-line jest/no-disabled-tests
+    it.skip('should be able to manage categories but not add new ones', async () => {
+      await goToAndExpandTaxonomyPanel();
```

Comment:
I think you should be able to enable this experiment using this util: https://github.com/google/web-stories-wp/blob/main/packages/e2e-test-utils/src/experimentalFeatures.js#L29

But we can always do this later in the issue that you created :)

Reply:
Handled in #9150